asserts/ifacedecls.go

```diff
@@ -148,16 +150,26 @@ func (matcher mapAttrMatcher) feature(flabel string) bool {
 	return false
 }
 
-func (matcher mapAttrMatcher) match(apath string, v interface{}, ctx AttrMatchContext) error {
+func (matcher mapAttrMatcher) match(attrer interfaces.Attrer, apath string, v interface{}, ctx AttrMatchContext) error {
+	// if we're starting at root path then use attrer to obtain top-level values
+	if apath == "" {
```

Comment:
it's a bit unclear to me why we need the extensive changes, it seems like this code could be lifted to AttributeContraints.Check, that's the only place where apath == "" if I'm rereading the code correctly